irteus/demo/hanoi.l

```diff
@@ -111,7 +111,8 @@
   (dotimes (i n)
     (setq disks (cons (hanoi-disk (+ (* (/ 27 n) i) 12) 20)
                       disks))
-    (send (car disks) :set-color (send (elt gl::*face-colors* (* (+ i 4) 32)) :name))
+    (when (memq :gl *features*)
+      (send (car disks) :set-color (send (elt gl::*face-colors* (* (+ i 4) 32)) :name)))
```

Comment:
This will cause read errors on `gl::*face-colors*` if the gl package is not defined. Using `#+:gl` might be a better choice.

Reply:
@Affonso-Gui Thanks for feedback.
1) Set `gl` as default package : In https://github.com/euslisp/EusLisp/pull/491/commits/9c5a6172ca6039a688e545d4cf62a3090ae756c2, I am defined `gl`  as default package, because "X" is already there. Of course, this breaks backward compatibility and will trouble if you do not update both euslisp and jskeus at same time..
2) Consider when compile source code with `eusg`: If we compile source code with `#+gl` with `eus2` or `eusg`, and load that `.so` file with `eusgl`, then the `#+gl` code is ignored. When we consider euslisp in Non-X environment, we want to compile  all code with `eusg`.
3) Afcourse, we will not compile `hanoi.l` with `eusg` and load with `eusgl`. I am ok with use `#+gl` at this section. But if your intend to use `#+gl` `#+xwindow` for other codes like https://github.com/euslisp/EusLisp/pull/487/files, we need find better solution.


````
$ bash ./my-test.sh
+ eus2 my-test.l
"done!!!"
+ eusgl my-test.l
"(memq :gl *features*)"
"#+:gl"
"done!!!"
+ eus2 '(progn (compile-file "my-test.l" :o "my-test-2.so") (exit 0))'
+ eusgl '(progn (compile-file "my-test.l" :o "my-test-gl.so") (exit 0))'
+ eus2 '(load "my-test-2.so" :entry "___my_test")'
"done!!!"
+ eus2 '(load "my-test-gl.so" :entry "___my_test")'
"#+:gl"
"done!!!"
+ eusgl '(load "my-test-2.so" :entry "___my_test")'
"(memq :gl *features*)"
"done!!!"
+ eusgl '(load "my-test-gl.so" :entry "___my_test")'
"(memq :gl *features*)"
"#+:gl"
"done!!!"
```

here is the code.
```
$ cat my-test.l 
(defun my-test-func nil
  (when (memq :gl *features*)
    (print "(memq :gl *features*)"))
#+:gl
  (print "#+:gl")
  (print "done!!!"))

(eval-when
 (eval load)
 (my-test-func)
 (exit 0))
$ cat my-test.sh
set -x

eus2  my-test.l 2> /dev/null
eusgl my-test.l 2> /dev/null

eus2  "(progn (compile-file \"my-test.l\" :o \"my-test-2.so\") (exit 0))"  2> /dev/null
eusgl "(progn (compile-file \"my-test.l\" :o \"my-test-gl.so\") (exit 0))" 2> /dev/null

eus2  "(load \"my-test-2.so\" :entry \"___my_test\")"  2> /dev/null
eus2  "(load \"my-test-gl.so\" :entry \"___my_test\")" 2> /dev/null

eusgl "(load \"my-test-2.so\" :entry \"___my_test\")"  2> /dev/null
```

Reply:
Interesting point. I was unnaware of the `#+/-` effect on compiled code. (Is this the same in common lisp?)

Adding gl to the default eus packages ( https://github.com/euslisp/EusLisp/commit/9c5a6172ca6039a688e545d4cf62a3090ae756c2  ) does 
seem like a nice solution for now! 

If the problem persists as we deal with minor / user-defined packages I guess we could also update the compiler entry on this. ( default to dynamic evaluation, force static with `#.`)